### ISSUE TITLE
MDX Distinct

### ISSUE THREAD

**Issue by jazzido** [rsim/mondrian-olap#32]
It would be great to have support for MDX `DISTINCT` operation in the query generator.

I'm trying to implement it, but I'm getting an error when running `rake db:create_data MONDRIAN_DRIVER=postgresql` for setting up my spec environment:

```
NoMethodError: undefined method `get_oid_type' for #<ActiveRecord::ConnectionAdapters::JdbcAdapter:0x3aeb625c>
/Users/manuel/.rvm/gems/jruby-1.7.19/gems/activerecord-jdbc-adapter-1.3.17/lib/arjdbc/postgresql/adapter.rb:1257:in `columns'
/Users/manuel/.rvm/gems/jruby-1.7.19/gems/activerecord-jdbc-adapter-1.3.17/lib/arjdbc/postgresql/adapter.rb:1251:in `columns'
/Users/manuel/.rvm/gems/jruby-1.7.19/gems/activerecord-4.2.3/lib/active_record/connection_adapters/schema_cache.rb:43:in `columns'
/Users/manuel/.rvm/gems/jruby-1.7.19/gems/activerecord-4.2.3/lib/active_record/attributes.rb:93:in `columns'
/Users/manuel/.rvm/gems/jruby-1.7.19/gems/activerecord-4.2.3/lib/active_record/attributes.rb:98:in `columns_hash'
/Users/manuel/.rvm/gems/jruby-1.7.19/gems/activerecord-4.2.3/lib/active_record/inheritance.rb:205:in `subclass_from_attributes?'
/Users/manuel/.rvm/gems/jruby-1.7.19/gems/activerecord-4.2.3/lib/active_record/inheritance.rb:54:in `new'
/Users/manuel/.rvm/gems/jruby-1.7.19/gems/activerecord-4.2.3/lib/active_record/persistence.rb:50:in `create!'
/Users/manuel/Work/cosas/mondrian-olap/spec/rake_tasks.rb:126:in `(root)'
/Users/manuel/Work/cosas/mondrian-olap/spec/rake_tasks.rb:125:in `(root)'
/Users/manuel/.rvm/gems/jruby-1.7.19/bin/jruby_executable_hooks:15:in `(root)'
```

**Comment by jazzido** [comment]
Got the specs running by locking ActiveRecord to `~> 4.0.0`

**Comment by jazzido** [comment]
I've implemented basic support the DISTINCT function: https://github.com/jazzido/mondrian-olap/commit/de8459a205439280a6035e690fd59aedabc32db9

It works for the type of queries that I'm generating for my application, but I'm not sure if the patch is general enough for other purposes. If so, feel free to merge or let me know and I'll send you a proper pull request.

Thanks for the amazing work that you've put in this package!

**Comment by jazzido** [comment]
Both changes are now available as pull requests #33 and #34. Thanks again!
